src/components/views/rooms/DecryptionFailureBar.tsx

```diff
@@ -154,78 +142,38 @@ export const DecryptionFailureBar: React.FC<IProps> = ({ failures }) => {
-                </React.Fragment>
-            );
-            button = (
-                <AccessibleButton
-                    className="mx_DecryptionFailureBar_end_button"
-                    kind="primary"
-                    onClick={onVerifyClick}
-                    data-testid="decryption-failure-bar-button"
-                >
-                    {_t("Verify")}
-                </AccessibleButton>
-            );
-        } else {
-            className = "mx_DecryptionFailureBar mx_DecryptionFailureBar--withEnd";
-            headline = <React.Fragment>{_t("Reset your keys to prevent future decryption errors")}</React.Fragment>;
-            message = (
-                <React.Fragment>
-                    {_t(
-                        "You will not be able to access old undecryptable messages, " +
-                            "but resetting your keys will allow you to receive new messages.",
-                    )}
-                </React.Fragment>
-            );
-            button = (
-                <AccessibleButton
-                    className="mx_DecryptionFailureBar_end_button"
-                    kind="primary"
-                    onClick={onResetClick}
-                    data-testid="decryption-failure-bar-button"
-                >
-                    {_t("Reset")}
-                </AccessibleButton>
-            );
-        }
-    } else if (hasOtherVerifiedDevices) {
+    } else if (needsVerification && (hasOtherVerifiedDevices || hasKeyBackup)) {
         className = "mx_DecryptionFailureBar mx_DecryptionFailureBar--withEnd";
-        headline = <React.Fragment>{_t("Open another device to load encrypted messages")}</React.Fragment>;
+        headline = <React.Fragment>{_t("Verify this device to access all messages")}</React.Fragment>;
         message = (
             <React.Fragment>
-                {_t(
-                    "This device is requesting decryption keys from your other devices. " +
-                        "Opening one of your other devices may speed this up.",
-                )}
+                {_t("This device was unable to decrypt some messages because it has not been verified yet.")}
             </React.Fragment>
         );
         button = (
             <AccessibleButton
                 className="mx_DecryptionFailureBar_end_button"
-                kind="primary_outline"
-                onClick={onDeviceListClick}
+                kind="primary"
+                onClick={onVerifyClick}
                 data-testid="decryption-failure-bar-button"
             >
-                {_t("View your device list")}
+                {_t("Verify")}
             </AccessibleButton>
         );
-    } else {
-        className = "mx_DecryptionFailureBar";
+        // In developerMode, we want to be able to resend manually key requests
+    } else if (developerMode && !needsVerification && hasOtherVerifiedDevices && anyUnrequestedSessions) {
```

Comment:
The comment should really be inside the relevant `if` clause:

```suggestion
    } else if (developerMode && !needsVerification && hasOtherVerifiedDevices && anyUnrequestedSessions) {
        // In developerMode, show a button to resend key requests if there are other devices we could request keys from.
```